lib/compiler/passes/generate-js.js

```diff
@@ -156,40 +156,40 @@ function generateJS(ast, options) {
     function buildRegexp(cls) {
       return "/^["
         + (cls.inverted ? "^" : "")
-        + cls.value.map(part => Array.isArray(part)
+        + cls.value.map(part => (Array.isArray(part)
           ? regexpClassEscape(part[0])
             + "-"
             + regexpClassEscape(part[1])
-          : regexpClassEscape(part)).join("")
+          : regexpClassEscape(part))).join("")
         + "]/" + (cls.ignoreCase ? "i" : "");
     }
 
-    function buildExpectation(e) {
-      switch (e.type) {
+    function buildExpectation(er) {
```

Comment:
`e` here is a shorthand for `expectation`, so it would be better to rename it to `expectation` / `exp`